platform/x11/tts_linux.cpp

```diff
@@ -75,69 +80,81 @@ void TTS_Linux::speech_init_thread_func(void *p_userdata) {
 
 void TTS_Linux::speech_event_index_mark(size_t p_msg_id, size_t p_client_id, SPDNotificationType p_type, char *p_index_mark) {
 	TTS_Linux *tts = TTS_Linux::get_singleton();
-	if (tts && tts->ids.has(p_msg_id)) {
-		MutexLock thread_safe_method(tts->_thread_safe_);
-		// Get word offset from the index mark injected to the text stream.
-		String mark = String::utf8(p_index_mark);
-		OS::get_singleton()->tts_post_utterance_event(OS::TTS_UTTERANCE_BOUNDARY, tts->ids[p_msg_id], mark.to_int());
+	if (tts) {
+		tts->call_deferred(PNAME("_speech_index_mark"), p_msg_id, p_client_id, (int)p_type, String::utf8(p_index_mark));
```

Comment:
Why `PNAME` here? The original had `SNAME` which is completely different, `PNAME` is for translation